x/evm/keeper/state_transition_benchmark_test.go

```diff
@@ -0,0 +1,134 @@
+package keeper_test
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/stretchr/testify/require"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+)
+
+var templateTx = &ethtypes.LegacyTx{
+	GasPrice: big.NewInt(1),
+	Gas:      21000,
+	To:       &common.Address{},
+	Value:    big.NewInt(0),
+	Data:     []byte{},
+}
+
+func newSignedEthTx(
+	txData *ethtypes.LegacyTx,
+	nonce uint64,
+	addr sdk.Address,
+	krSigner keyring.Signer,
+	ethSigner ethtypes.Signer,
+) (*ethtypes.Transaction, error) {
+	txData.Nonce = nonce
+	ethTx := ethtypes.NewTx(txData)
+
+	sig, _, err := krSigner.SignByAddress(addr, ethTx.Hash().Bytes())
+	if err != nil {
+		return nil, err
+	}
+
+	ethTx, err = ethTx.WithSignature(ethSigner, sig)
+	if err != nil {
+		return nil, err
+	}
+
+	return ethTx, nil
+}
+
+func newNativeMessage(
+	nonce uint64,
+	blockHeight int64,
+	address common.Address,
+	cfg *params.ChainConfig,
+	krSigner keyring.Signer,
+	ethSigner ethtypes.Signer,
+) (core.Message, error) {
+	msgSigner := ethtypes.MakeSigner(cfg, big.NewInt(blockHeight))
+
+	templateTx.Nonce = nonce
+	ethTx := ethtypes.NewTx(templateTx)
```

Comment:
ideally we should have benchmarks for each of the tx data types, and default to the latest one

Reply:
added both tx type